### DISCUSSION TITLE
Managing articles ("a", "an", "the") and last name sorting

### DISCUSSION THREAD

**Discussion by kylebarbour** [beetbox/beets#5041] [Q&A] [answered]
Hello!

I'm new to beets and trying to figure out how best to use it. I prefer my library sorted in the style of a standard index, moving articles to the end (e.g. "Flashbulb, The" and not "The Flashbulb") and sorting by last name (e.g. "Cave, Nick" and not "Nick Cave"). Ideally, this would be the case both in their tags and paths.

What is the canonical way to address these goals in beets? Initially, I tried using 

    paths:
      default: $albumartist_sort/$album%aunique{}/$track - $title

but this results in many artists being filed under `_`, presumably because `albumartist_sort` is undefined. I discovered the [the plugin](https://beets.readthedocs.io/en/stable/plugins/the.html), which is helpful for paths, but doesn't affect the tags. I already use the [edit plugin](https://beets.readthedocs.io/en/stable/plugins/edit.html), but this is very tedious and requires manually editing every track during import mode, or logging all of the incorrectly tagged albums and manually going through them in album mode. It doesn't appear that MusicBrainz has separate last and first name fields, which further complicates the matter.

I imagine this sorting strategy is desired by other people &mdash; if so, how have you solved these issues? Intuitively to me, using `albumartist_sort` seems the cleanest, but I can't get it to work consistently.

Thanks!

**Comment by DjSlash** [comment, marked as answer]
As far as I know, `$albumartist_sort` should be populated, but it might depend on the data source. Either way, to make sure that it is set and usable in your path, you can use something like:

```yaml
paths:
  default: %if($albumartist_sort, $albumartist_sort, $albumartist)/$album%aunique{}/$track - $title
```

As for tags, your player should be able to use the tag `ALBUMARTISTSORT` from the file.

**Reply by kylebarbour** [reply]
I'll play around with this, thanks!
